portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,16 +48,25 @@ public Comment(String body, String author, Date date) {
         }
     }
 
-    /** GETs all comments stored by the server */
+    /** GETs a user-defined number of comments stored by the server */
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+        // Get the input from the form.
+        int numberOfCommentsToShow = 10;
+        if (request.getParameter("num-comments") != null && !request.getParameter("num-comments").isEmpty()){
+            try {
+                numberOfCommentsToShow = Integer.parseInt(request.getParameter("num-comments"));
+            } catch (Exception e) {}
```

Comment:
Handle the exception here. You can add a TODO for now. That's okay. From Effective Java

> An empty catch block defeats the purpose of exceptions, which is to force you to handle exceptional conditions. Ignoring an exception is analogous to ignoring a fire alarm—and turning it off so no one else gets a chance to see if there’s a real fire. You may get away with it, or the results may be disastrous. Whenever you see an empty catch block, alarm bells should go off in your head. At the very least, the catch block should contain a comment explaining why it is appropriate to ignore the exception.

Reply:
Sounds good! My thinking was that if the user passes something that can't be parsed as an integer to the endpoint, the server should just silently fall back on the default value assigned to the variable (10), instead of stopping the flow of execution. Would it be acceptable to move the assignment of `numberOfCommentsToShow = 10` into the exception block or is that also an anti-pattern?